### ISSUE TITLE
Enhancement: Adjust Level-2 elements. Add border-3 and dark theme foreground change

### ISSUE THREAD

**Issue by bstifle** [Esri/calcite-components#1431]
In order to keep 1:1 change between light and dark theme, these changes have been vetted by design group:

![image](https://user-images.githubusercontent.com/14083530/103248827-f81ff080-4931-11eb-8fff-738f8ad595f9.png)

- Change border radius to 4px `--calcite-border-radius: 4px`
- Add `border-3` to light theme level 2 components
- Add `border-3-dark` to dark theme level 2 components
- Use `foreground-1-dark` on dark theme level 2 components instead of foreground-3

This applies to dropdown, alert, modal, date picker, tooltip, popover, color picker (when it's a dropdown)

**Comment by caripizza** [comment]
@bstifle does this one still apply?

**Comment by bstifle** [comment]
Yes i believe so!

**Comment by julio8a** [comment]
checking in on this one again

**Comment by caripizza** [comment]
@julio8a @bstifle isn't this already covered in our refactors based on the Figma designs?

**Comment by julio8a** [comment]
Yes, closing.

**Comment by bstifle** [comment]
Hmm. Not seeing these shadows in our components @caripizza @julio8a

**Comment by caripizza** [comment]
@bstifle do the shadows appear in the Figma designs that we're using for the refactors?

**Comment by bstifle** [comment]
@caripizza yep! i wonder if this is just a shadows utility class we need to fix? other than the shadows, the other design criteria seem to be met in the issue

**Comment by caripizza** [comment]
Re-opened after clarifying the remaining work needed in this issue with Bryan (@bstifle please correct or add anything I missed from our chat).

Based on first glance, it looks like:

1. [_popper.scss](https://github.com/Esri/calcite-components/blob/master/src/assets/styles/_popper.scss#L19) global styles - currently using `$shadow-2` sass var - needs to instead use `@apply shadow-2`: 
2. Additionally, the `%shadow-vars` sass placeholder we have here: [src/assets/styles/_shadow.scss](https://github.com/Esri/calcite-components/blob/master/src/assets/styles/_shadow.scss#L1)
Don't seem to be used right now. We don't want these as global CSS Vars as far as I know, so let's remove them. As long as we have what we need in TW config under [`boxShadow`](https://github.com/Esri/calcite-components/blob/master/tailwind.config.js#L138) object, we can refer to those moving forward. (But why do we have duplicate 0 and 1 values?)

@julio We should also use this issue to update the shadows content [in our readme](https://github.com/Esri/calcite-components/blob/master/src/assets/styles/readme.md#shadows) to make sure it corresponds with (or points to) the content we add in dev docs:
https://github.com/ArcGIS/afd-calcite-documentation/issues/57

**Comment by caripizza** [comment]
Can we update the description to a) articulate which components are Level 0, 1, or 2 (I don't know), and b) so we can style their shadows based on their level?

**Comment by bstifle** [comment]
@caripizza this issue is for specific components (listed in the description at the bottom) and wont apply to all components with shadows

**Comment by bstifle** [comment]
here are the depth values and which components use them:

<img width="1250" alt="Depth Light" src="https://user-images.githubusercontent.com/14083530/127207833-4698ae6c-a166-49ce-98cc-52d371e16a5d.png">
<img width="1250" alt="Depth Dark" src="https://user-images.githubusercontent.com/14083530/127207842-de7b0085-25fe-42d3-8591-ba6da28d5493.png">

**Comment by julio8a** [comment]
This issue needs another design review (Figma) before moving into code.

**Comment by caripizza** [comment]
Should the updated box-shadows correspond to the output of this calcite-base mixin?

https://github.com/Esri/calcite-base/blob/master/dist/_shadow.scss

(Currently only referenced in [calcite-slider](https://github.com/Esri/calcite-components/blob/master/src/components/calcite-slider/calcite-slider.scss#L256).)

**Comment by julio8a** [comment]
This one was moved to the Oct 11 sprint. @macandcheese will need to give this another design review.

**Comment by caripizza** [comment]
Moving this issue to the Freezer after discussion with @macandcheese.

**Comment by asangma** [comment]
The proposed shadows are shown in [this codepen](https://codepen.io/mtb-alan/pen/mdXWpJo).

It appears that except for `shadow-1` and  `shadow-2`, the other shadows are only used in these cases:

- shadow-1-sm
  - Card's active state. Card shouldn't have an active state since it's not clickable.
- shadow-1-lg
  - Card's hover state. Card shouldn't have an active state since it's not clickable.
- shadow-2-sm
  - Fab's active state
  - modal
 - shadow-2-lg
  - Fab's hover state

Also, it seems that shadow-0 and shadow-1 in the tailwind confige are the same?

Propose removing the rarely use shadows and updating those components as such:

- Card active and hover states would not exist.
- Fab's states would be handled with color states that match Button
- Modal could use 1.

**Comment by macandcheese** [comment]
Agree w @asangma assessment - these could be greatly simplified. Let's also consider other theming options for shadow so even if the "default Calcite" theme doesn't have a shadow in the design, an overriding theme may.

**Comment by geospatialem** [comment]
Prioritization will be made after the 1.0 release next month.

**Comment by geospatialem** [comment]
Related issue https://github.com/Esri/calcite-components/issues/3122

**Comment by macandcheese** [comment]
@ashetland @SkyeSeitz proposing we close this one and address in an updated issue if this is still needed, but this seems a bit outdated at this point compared to current Figma.

**Comment by ashetland** [comment]
Agreed. Would be dealt with in #3122.

**Comment by SkyeSeitz** [comment]
> Agreed. Would be dealt with in #3122.

+1
